gems/aws-crt/native/src/signing.c

```diff
@@ -0,0 +1,16 @@
+#include "crt.h"
+
+#include <aws/auth/credentials.h>
+#include <aws/auth/signing.h>
+#include <aws/auth/signing_config.h>
+#include <aws/common/string.h>
+
+int aws_crt_sign_request(
+    const struct aws_signable *signable,
+    const struct aws_crt_signing_config *config,
+    const char *sign_id,
+    aws_signing_complete_fn *on_complete) {
+
+    return aws_sign_request_aws(
+        aws_crt_allocator(), signable, (struct aws_signing_config_base *)config, on_complete, (void *)sign_id);
```

Comment:
Yeah, this is a good catch.  
I've removed the string and am just passing NULL for the userdata.

I had initially wanted to create the `aws_crt_sign_request_synchronous` function to avoid having the Ruby callback (and therefore crossing the language boundry an extra time).  However, the `on_complete` function callback doc states "result will be destroyed after this function returns" - which means we would need to copy that data.  And then the Ruby code would need to copy that data again (the copies that are currently done in the callback).  So its a tradeoff between an extra copy or a callback.  For now, I've kept the callback.